src/dos/drive_local.cpp

```diff
@@ -172,17 +174,19 @@ bool localDrive::FileOpen(DOS_File **file, const char *name, uint8_t flags)
 
 	if (file_handle == InvalidNativeFileHandle) {
 		DOS_SetError(DOSERR_INVALID_HANDLE);
-		return false;
+		return nullptr;
 	}
 
 	if (fallback_to_readonly) {
 		MaybeLogFilesystemProtection(host_filename);
 	}
 
-	*file = new localFile(name, host_filename, file_handle, basedir, IsReadOnly());
-	(*file)->flags = flags;  // for the inheritance flag and maybe check for others.
+	auto file = std::make_unique<localFile>(
+	        name, host_filename, file_handle, basedir, IsReadOnly());
+	file->flags = flags; // for the inheritance flag and maybe check for
+	                     // others.
```

Comment:
minor: comment on line above while you're at it 😄

Reply:
I already removed this comment in a later commit.